### ISSUE TITLE
Download 5.22.0 when .travis.yml asks for '5.22' and similar

### ISSUE THREAD

**Issue by autarch** [travis-perl/helpers#25]
It'd be nice if the helpers were smart enough to just download the latest "X.Y.Z" when the config asks for "X.Y".

I'll try to get to this if I find some round tuits, but feel free to beat me to it ;)

**Comment by haarg** [comment]
The reason I don't do this currently is that I didn't want to change the behavior of the existing version targets travis provides.  But if for example, you request 5.12-thr, it will give you a 5.12.5 build with threads enabled.

Do you think it would be better to always try to select the most recent version, even if travis provides an existing version under the same name?

**Comment by autarch** [comment]
Good question. Here's how I think this should all work ...

If someone asks for a specific patchlevel, like `5.22.0`, they should get exactly that. If possible, it should come from the Travis environment (since that's faster).

If someone asks for just a major/minor version, like `5.22`, they should get the first of ...

* Perl from Travis that is 5.22.X. If there are multiple patchlevels available, use the most recent.
* Pre-built Perl from travis-perl. If there are multiple patchlevels available, use the most recent.

Basically, I'm saying it should optimize for speed, and if someone always wants the latest version of a given release, they should ask for it explicitly.
